src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Wasi.cs

```diff
@@ -0,0 +1,154 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using static Interop;
+using static Interop.Sys;
+
+namespace System.Net.Sockets
+{
+    internal sealed unsafe class SocketAsyncEngine
+    {
+        internal static readonly bool InlineSocketCompletionsEnabled = Environment.GetEnvironmentVariable("DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS") == "1";
```

Comment:
I don;t think you need this. This is related to thread pool optimization and there are no threads on WASI, right?

Reply:
There is `InlineSocketCompletionsEnabled` -> `SafeSocketHandle.PreferInlineCompletions` -> `context.PreferInlineCompletions` which needs it. 

I don't understand why this is not a static member on `SafeSocketHandle`. Do we expect the env variable to change mid-flight ?

For WASI I made it constant on `SocketAsyncEngine`